tv_apps/fling-player/js/fling_player.js

```diff
@@ -262,58 +283,24 @@
     this.setPlayButtonState('paused');
   };
 
-  proto._startUpdateControlPanelContinuously = function () {
-
-    if (this._contrlPanelUpdateTimer == null) {
+  proto._updateControlPanel = function () {
+    var buf = this._player.getVideo().buffered;
+    var current = this._player.getRoundedCurrentTime();
 
-      // To stop this is important.
-      // We don't want they get messed with each other
-      this._stopSeekOnKeyPress();
+    this.writeTimeInfo('elapsed', current);
+    this.moveTimeBar('elapsed', current);
 
-      this._contrlPanelUpdateTimer = setTimeout(
-        this._updateControlPanelContinuously.bind(this)
-      );
-      this.showControlPanel(true);
-    }
-  };
-
-  proto._stopUpdateControlPanelContinuously = function () {
-    clearTimeout(this._contrlPanelUpdateTimer);
-    this._contrlPanelUpdateTimer = null;
-  };
-
-  /**
-   * This is to keep auto updating the info and status on the control panel.
-   */
-  proto._updateControlPanelContinuously = function () {
-
-    if (this._contrlPanelUpdateTimer != null) {
-
-      var buf = this._player.getVideo().buffered;
-      var current = this._player.getRoundedCurrentTime();
-
-      this.writeTimeInfo('elapsed', current);
-      this.moveTimeBar('elapsed', current);
-
-      if (buf.length) {
-        this.moveTimeBar('buffered', buf.end(buf.length - 1));
+    for (var bufEnd, bufStart, i = 0; i < buf.length; ++i ) {
+      bufEnd = buf.end(i);
+      bufStart = buf.start(i);
+      if (bufStart <= current && current <= bufEnd) {
+        this.moveTimeBar('buffered', bufEnd);
```

Comment:
I'm not sure about the situation containing multiple `buffered`, do you have any example? Should we do something like choosing the latest time?